weave/trace/serialization/custom_objs.py

```diff
@@ -47,25 +48,61 @@ def encode_custom_obj(obj: Any) -> dict | None:
         # Ensure load_instance is an op
         if not isinstance(serializer.load, Op):
             serializer.load = op(serializer.load)
-        # Save the load_intance_op
+        # Save the load_instance_op
         wc = require_weave_client()
 
         # TODO(PR): this can fail right? Or does it return None?
         # Calculating this URL is blocking, but we only have to pay it once per custom type
         load_instance_op_ref = wc._save_op(serializer.load, "load_" + serializer.id())  # type: ignore
         load_op_uri = load_instance_op_ref.uri()
 
-    encoded_path_contents = {
-        k: (v.encode("utf-8") if isinstance(v, str) else v)  # type: ignore
-        for k, v in art.path_contents.items()
-    }
-    return {
+    encoded = {
         "_type": "CustomWeaveType",
         "weave_type": {"type": serializer.id()},
-        "files": encoded_path_contents,
         "load_op": load_op_uri,
     }
 
+    # If the save method just takes one argument, it is an inline serializer
+    save_signature = inspect.signature(serializer.save)
+    param_count = len(save_signature.parameters)
```

Comment:
hmm I don't love this but maybe ok for now.

Longer term I think it would be better to type this literally as an inline vs. artifact serializer

Reply:
I'll make the typing a little stricter and add some typeguard methods for readability. Doing it properly would require more of a change than I'd like to do right now because of circular imports.